### PR TITLE
v1.2. Implement FilterDepartmentCommand and SortCommand

### LINKED ISSUE #37: As a user, I want to sort my address book by name and department
so that I can look for contacts that I want easily.

### LINKED ISSUE #39: As a user, I can filter out and view my relevant colleague's department
so that I can spend less time searching for their contact details and contact them.

### PR DESCRIPTION
**Pull request by Woonhian**
**v1.2**

### SortCommand
- Added `sort` command
- Able to sort employees based on their name or sort by departments in ascending or descending order

### FilterDepartmentCommand
- Added `filterdepartment` command
- Alias: `fd`
- Able to filter out departments to view certain department employee list
- The word 'Management' will not be able to filter departments to prevent the command from showing all of the departments as all departments have the word 'Management'

#### Enhancements
- Changed department fields naming to be strict such that naming of a department needs to be a alphabetic name, followed by 'Management'. e.g. Junior Management

**Test cases added for SortCommand and FilterDepartmentCommand**

Resolves #37 
Resolves #39

### CODE REVIEW

**Review by pinjuen — COMMENTED** [review]
useful for searching by department!

File: src/main/java/seedu/address/model/person/Department.java [hunk 1]
```diff
@@ -10,13 +10,14 @@
...
-            "Department names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Department names should only contain alphabetic characters and spaces, and it should not be blank\n"
+            + "Department names should start with a name, and ends with 'Management'";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String DEPARTMENT_VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String DEPARTMENT_VALIDATION_REGEX = "(\\p{Alpha}+ )+Management";
```
Comment (Hafizuddin-github) [inline src/main/java/seedu/address/model/person/Department.java:21]: Good enhancement!

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 2]
```diff
@@ -102,6 +105,59 @@ public void setPersons(List<Person> persons) {
...
+     * @return
+     */
+    public void sortBy(String field, String order) throws NoEmployeeException {
+        if (internalList.size() < 1) {
+            throw new NoEmployeeException();
+        }
+
+        Comparator<Person> comparator = null;
+
+        Comparator<Person> nameComparator = new Comparator<Person>() {
```
Comment (jinyao-lee) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:123]: Might want to add comment for the two lambda functions

**Comment by jylee-git** [comment]
PR approved.

**Review by jinyao-lee — COMMENTED** [review]
PR approved.

**Pull request merged**